application/classes/Ushahidi/Console/PostExporter.php

```diff
@@ -131,34 +136,47 @@ protected function executeExport(InputInterface $input, OutputInterface $output)
 			if ($job->filters) {
 				$filters = array_merge($filters, $job->filters);
 			}
-			
+
 			// Set the fields that should be included if set
 			if ($job->fields) {
 				$data->include_attributes = $job->fields;
 			}
 		}
 
-        foreach ($filters as $key => $filter) {
-            $data->$key = $filter;
+		foreach ($filters as $key => $filter) {
+			$data->$key = $filter;
 		}
 
-        $this->postExportRepository->setSearchParams($data);
-		
-        $posts = $this->postExportRepository->getSearchResults();
+		$this->postExportRepository->setSearchParams($data);
+		$posts = $this->postExportRepository->getSearchResults();
+		service("formatter.entity.post.$format")->setFileSystem($this->fs);
+		service("formatter.entity.post.$format")->setAddHeader($add_header ===  'true');
+		//fixme add post_date
+		$form_ids = $this->postExportRepository->getFormIdsForHeaders();
+		$attributes = $this->formAttributeRepository->getByForms($form_ids);
+
+		$keyAttributes = [];
+		foreach($attributes as $key => $item)
+		{
+			$keyAttributes[$item['key']] = $item;
+		}
 
 		// // ... remove any entities that cannot be seen
 		foreach ($posts as $idx => $post) {
-
 			// Retrieved Attribute Labels for Entity's values
-			$post = $this->postExportRepository->retrieveColumnNameData($post->asArray());
-
+			$post = $this->postExportRepository->retrieveMetaData($post->asArray(), $keyAttributes);
```

Comment:
Isn't this going to be pretty horribly inefficient? ie. reload attributes etc for *every* post?
Are we caching somewhere? or can we do this in aggregate somehow?

Reply:
Good point. Follow up PR? Or just backlog a tech debt task?

Reply:
follow up PR for sure. I'd rather solve it while it's fresh. But I'll open myself a ticket because otherwise It'll get lost in the morning